### PR TITLE
Relay commands

### PR DESCRIPTION
**Pull request by mpeck**
Adds support for listing relays and updating their name and/or description.

Would really appreciate thoughts on the help bits. I rolled usage info into the command in order to get more control.

reference #513 

Note: I couldn't get the display output exactly like I wanted it. I think we will need a few updates to FuMachu before we can easily render tables without piping into the table command. Created a ticket for that here, #611.

### CODE REVIEW

**Comment by kevsmith** [comment]
Couple of UX issues I noticed using the relay command:

- `relay list` when no relays are installed returns "Pipeline executed successfully but no output was returned". The message should be "No relays configured".
- `relay update` should allow the user to update the relay's token too.

**Comment by kevsmith** [comment]
How will the user get help for the subcommands?

File: lib/cog/commands/relay/list.ex [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      if Helpers.flag?(options, "group") do
+        Map.put(relay_map, "relay_groups", Enum.map(relay.groups, &generate_group_map/1))
+        |> Map.put("_show_groups", true)
+      else
+        relay_map
+      end
+    end)
+  end
+
+  defp verbose_relay(relay) do
```
Comment (kevsmith) [inline lib/cog/commands/relay/list.ex]: I'd unify `verbose_relay/1` and `standard_relay/1` into a single function that returned all the data. You can then handle what the user sees in the output template without impacting usefulness within a pipeline.

**Comment by mpeck** [comment]
@kevsmith the token update bit was specifically called out to not expose in the original issue. Is that no longer the case?

**Comment by mpeck** [comment]
And also, I'm glad you pointed out the "Pipeline executed successfully but no output was returned" thing. We get that from the executor. If you return an empty list the executor bypasses templating and renders that message. Is that desired behavior? It does kinda suck since you can't handle 'no results' cases in your template.

File: lib/cog/commands/relay/list.ex (hunk 1 above)
Comment (mpeck) [inline lib/cog/commands/relay/list.ex]: So I actually thought about that @kevsmith, and originally that's how I planned on doing it. But there isn't a way to easily get additional metadata into the template without polluting the record. I could just stick flags in each object to turn things on and off, that's actually what I had to do with the relay groups bit to get it to display properly, but that seems like a generally bad idea. Do you have any thoughts?

**Comment by kevsmith** [comment]
@mpeck Thanks for pointing out the token bit in the original issue. I _assumed_ we'd want parity with `cogctl` but it makes total sense we wouldn't.

**Comment by kevsmith** [comment]
Re: no output -- possibly we could handle with a template by emitting `[{relays: []}]` and specifying an empty result template?

**Pull request merged**